proxy/middleware.go

```diff
@@ -152,50 +159,60 @@ func SetResponseHeaders(headers map[string]string) func(next http.Handler) http.
 	}
 }
 
-func (p *Proxy) jwtClaimMiddleware(next http.Handler) http.Handler {
-	return httputil.HandlerFunc(func(w http.ResponseWriter, r *http.Request) error {
-		if jwt, err := sessions.FromContext(r.Context()); err == nil {
-			var jwtClaims map[string]interface{}
-			if err := p.encoder.Unmarshal([]byte(jwt), &jwtClaims); err == nil {
-				formattedJWTClaims := make(map[string]string)
-
-				// reformat claims into something resembling map[string]string
-				for claim, value := range jwtClaims {
-					var formattedClaim string
-					if cv, ok := value.([]interface{}); ok {
-						elements := make([]string, len(cv))
-
-						for i, v := range cv {
-							elements[i] = fmt.Sprintf("%v", v)
+// jwtClaimMiddleware logs and propagates JWT claim information via request headers
+//
+// if returnJWTInfo is set to true, it will also return JWT claim information in the response
+func (p *Proxy) jwtClaimMiddleware(returnJWTInfo bool) mux.MiddlewareFunc {
+
+	return func(next http.Handler) http.Handler {
+
+		return httputil.HandlerFunc(func(w http.ResponseWriter, r *http.Request) error {
+			if jwt, err := sessions.FromContext(r.Context()); err == nil {
+				var jwtClaims map[string]interface{}
+				if err := p.encoder.Unmarshal([]byte(jwt), &jwtClaims); err == nil {
+					formattedJWTClaims := make(map[string]string)
```

Comment:
could we pull this into a separate function? Like 

```go
func getFormattedJWTClaims(rawJWT []byte) (map[string]string, error)
```

Reply:
Yes.  I kept staring at that code and you're right.  Will break it apart a bit for clarity.